### PR TITLE
Inline mobile category and group edit

### PR DESCRIPTION
**Pull request by joel-jeremy**
Inline mobile category / group edit

### CODE REVIEW

**Comment by youngcw** [comment]
A few comments

1. Probably should only allow one edit at a time.  Currently I can open the edit view for every category at once.
2. I think the 3 dot menu should be outside that text box.  Currently to me it implies that the menu is only has name options, not the hide or delete options.

**Comment by joel-jeremy** [comment]
@youngcw I pushed an update to address your comments :)

**Comment by shall0pass** [comment]
It almost feels too easy to enter the editing of category names.  It's not something I would consider doing on a daily basis.  Is there any way we can lock/unlock the ability to edit the names/hide/unhide so that it doesn't become distracting?

**Comment by youngcw** [comment]
I had wondered about it being to easy to open the edit menu.  Would it still make sense if you had to press and hold to open the edit view?

**Comment by shall0pass** [comment]
> I had wondered about it being to easy to open the edit menu. Would it still make sense if you had to press and hold to open the edit view?

I think a long press could feel right.

**Comment by youngcw** [comment]
I like the long press.  I think it feels natural enough since those settings don't change often.

**Comment by joel-jeremy** [comment]
For some reason the long press does not work with Android Chrome. Instead of showing the category name input, it just shows the tap-to-search feature.

**Comment by shall0pass** [comment]
I'm seeing mixed success with brave.  Sometimes I can press away to close, sometimes it won't close unless I press over to an area like the spent column, and sometimes it selects the text.  Other times it works great....

**Comment by shall0pass** [comment]
Whatever you just did really fixed it for me.  I think this interaction feels natural.  Really cool!

**Comment by youngcw** [comment]
Still broken Android Brave

**Comment by joel-jeremy** [comment]
It also doesn't work for me in Chome and Brave.

**Comment by youngcw** [comment]
It kinda works on android firefox, but you cant hold too long or it selects the text after opening the edit view.

**Comment by youngcw** [comment]
I like the hold to select but its hard to accidentally select the budget value for editing, and I expect the category will be harder to accidentally open.  Maybe the best thing would be to not do the hold to open and see how it goes.  I bet for right handers it won't be an issue at all, and for lefties only occasionally an issue.

**Comment by joel-jeremy** [comment]
Agreed. I have reverted it to be on tap and get some feedback when it's released since it's tricky to make it work consistently on all mobile browsers.

**Comment by joel-jeremy** [comment]
Also brought back the "edit mode" where all groups/categories are made editable. I think this is how the mobile app used to do editing.

**Comment by youngcw** [comment]
On android brave the category name is getting highlighted on press.  So a single press on the category/group name opens the edit view and also opens up a system copy/paste dialog.

**Comment by joel-jeremy** [comment]
I get the same behavior but I don't mind it at all. It highlights the entire category name on touch which makes it easy to delete the entire category name and type a different name.

**Comment by shall0pass** [comment]
I'm seeing a bug when using my phone.

https://github.com/actualbudget/actual/assets/20625555/acc30bce-5f3c-446e-b244-a417f8911fb0

If I touch away after editing the field, it remembers the value I typed in.  I would expect this to either forget what I wrote or apply what I wrote.  I don't see this on a PC when using a mouse.

**Comment by joel-jeremy** [comment]
@shall0pass The category name edit issue should now be fixed :)

**Comment by shall0pass** [comment]
> @shall0pass The category name edit issue should now be fixed :)

Indeed!  This is much improved.  Well done!

File: packages/desktop-client/src/hooks/useLongPress.ts
Comment (youngcw) [inline packages/desktop-client/src/hooks/useLongPress.ts:1]: Is this file still being used?  We decided against the long press and I cant find this being called anywhere

Comment (joel-jeremy) [reply packages/desktop-client/src/hooks/useLongPress.ts:1]: It's no longer used. Deleted the file :)

**Review by youngcw — APPROVED** [review]

**Pull request merged**